src/rust/engine/process_execution/src/remote_cache.rs

```diff
@@ -256,19 +256,24 @@ impl CommandRunner {
       .ok_or_else(|| format!("File {:?} did not exist locally.", file_path))
   }
 
-  async fn make_action_result(
+  pub(crate) async fn make_action_result(
     &self,
     command: &Command,
     result: &FallibleProcessResultWithPlatform,
     store: &Store,
-  ) -> Result<ActionResult, String> {
+  ) -> Result<(ActionResult, Vec<Digest>), String> {
+    // Keep track of digests that need to be uploaded.
```

Comment:
Perhaps add a comment explaining that this may contain both File Digests and Directory Digests. (If those are the correct terms). I could see that not being immediately obvious. Perhaps have that in the docstring to describe the return type.

Reply:
Done. It actually includes Tree digests and not Directory digests.